### PR TITLE
quell type and immutable deprecation warnings under 0.7

### PR DESCRIPTION
**Pull request by Sacha0**
... and raise the minimum Julia version to 0.6. Best!

### CODE REVIEW

**Comment by stevengj** [comment]
I'd prefer to wait a little while, say to the end of summer, before dropping 0.5.

**Comment by Sacha0** [comment]
Sounds good! :)

**Comment by stevengj** [comment]
Needs a rebase?

**Comment by Sacha0** [comment]
Conflict (hopefully) resolved. Thanks!

File: src/colormaps.jl [hunk 1]
```diff
@@ -7,7 +7,7 @@ export ColorMap, get_cmap, register_cmap, get_cmaps
 ########################################################################
 # Wrapper around colors.Colormap type:
 
-type ColorMap
+mutable struct ColorMap
```
Comment (stevengj) [inline src/colormaps.jl]: I don't think this needs to be mutable, actually.

File: src/PyPlot.jl [hunk 2]
```diff
@@ -11,7 +11,7 @@ using Compat
 import Base.show
 
 # Wrapper around matplotlib Figure, supporting graphics I/O and pretty display
-type Figure
+mutable struct Figure
```
Comment (stevengj) [inline src/PyPlot.jl]: Probably just `struct`; similarly below

**Comment by PallHaraldsson** [comment]
Just close this as 0.6 is now minimum? Or something else here that needs to be merged?

**Comment by Sacha0** [comment]
Yes, this pull request is about as stale and obviated as it comes so let's close! :)

**Pull request closed without merging**